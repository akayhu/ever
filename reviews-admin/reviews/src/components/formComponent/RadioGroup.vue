<template>
  <div class="radio-option-lists">
    <el-radio-group
      :key="i"
      v-for="(company, i) in companyLists"
      @input="handleChange(company.name, company.custno, company.hasReview)"
    >
      <el-radio :label="company.name">{{ company.name }}</el-radio>
    </el-radio-group>
  </div>
</template>

<script>
export default {
  data() {
    return {
      selectedValue: null
    };
  },
  props: {
    companyLists: {
      type: Array,
      required: true
    },
    value: {
      type: String,
      default: null
    },
    validation: {
      type: Object,
      default: null
    }
  },
  mounted() {
    this.setDataGtm();
  },
  methods: {
    setDataGtm() {
      const allRadioInput = document.querySelectorAll(
        ".companyLayer .el-radio__original"
      );
      allRadioInput.forEach(radio => {
        radio.setAttribute("data-gtm-form", "選取多筆公司");
      });
    },
    handleChange(compantyName, custno, hasReview) {
      if (this.validation) this.validation.$touch();
      this.selectedValue = compantyName;
      this.$emit("value-changed", compantyName, custno, hasReview);
    }
  },
  computed: {
    // selectedValue() {
    //   return this.value;
    // }
  }
};
</script>

<style lang="scss">
.radio-option-lists {
  .el-radio-group {
    width: 100%;
    font-weight: 400;
  }
}
</style>
