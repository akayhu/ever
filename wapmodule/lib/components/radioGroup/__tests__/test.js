'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactTestRenderer = require('react-test-renderer');

var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);

var _reactAddonsTestUtils = require('react-addons-test-utils');

var _reactAddonsTestUtils2 = _interopRequireDefault(_reactAddonsTestUtils);

var _index = require('../index.js');

var _index2 = _interopRequireDefault(_index);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var group = [{ label: '項目1', value: '111' }, { label: '項目2', value: '222' }, { label: '項目3', value: '333' }, { label: '項目4', value: '444' }, { label: '項目5', value: '555' }];

/*import React from 'react/addons';
import RadioGroup from '../index.js';

const chai = require("chai");
const expect = chai.expect;



module.exports = describe("React Tesrt", () => {
	 it("should render something", () => {  
		let foo = React.addons.TestUtils.renderIntoDocument(<RadioGroup group={group}/>);
		expect(foo.getDOMNode()).to.exist; // 看有沒有 render 出 DOM
	});
})*/

describe('Radio', function () {
	var component = _reactTestRenderer2.default.create(_react2.default.createElement(_index2.default, { group: group }));

	beforeEach(function () {});

	it('initial component test', function () {

		var tree = component.toJSON();
		expect(tree).toMatchSnapshot();

		var radioElement = TestUtils.scryRenderedDOMComponentsWithClassName();
		expext(component.scryRenderedDOMComponentsWithTag);
	});
});