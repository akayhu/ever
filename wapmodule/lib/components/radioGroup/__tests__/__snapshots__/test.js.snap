exports[`Radio initial 1`] = `
<div
  className="radioGroup">
  <div
    className="radioItem">
    <input
      defaultChecked={null}
      id="undefinedradio0"
      label="項目1"
      name={undefined}
      onChange={[Function bound handleChange]}
      type="radio"
      value="111" />
    <label
      htmlFor="undefinedradio0"
      onClick={[Function bound handleClick]}>
      <div
        className="check" />
      項目1
    </label>
  </div>
  <div
    className="radioItem">
    <input
      defaultChecked={null}
      id="undefinedradio1"
      label="項目2"
      name={undefined}
      onChange={[Function bound handleChange]}
      type="radio"
      value="222" />
    <label
      htmlFor="undefinedradio1"
      onClick={[Function bound handleClick]}>
      <div
        className="check" />
      項目2
    </label>
  </div>
  <div
    className="radioItem">
    <input
      defaultChecked={null}
      id="undefinedradio2"
      label="項目3"
      name={undefined}
      onChange={[Function bound handleChange]}
      type="radio"
      value="333" />
    <label
      htmlFor="undefinedradio2"
      onClick={[Function bound handleClick]}>
      <div
        className="check" />
      項目3
    </label>
  </div>
  <div
    className="radioItem">
    <input
      defaultChecked={null}
      id="undefinedradio3"
      label="項目4"
      name={undefined}
      onChange={[Function bound handleChange]}
      type="radio"
      value="444" />
    <label
      htmlFor="undefinedradio3"
      onClick={[Function bound handleClick]}>
      <div
        className="check" />
      項目4
    </label>
  </div>
  <div
    className="radioItem">
    <input
      defaultChecked={null}
      id="undefinedradio4"
      label="項目5"
      name={undefined}
      onChange={[Function bound handleChange]}
      type="radio"
      value="555" />
    <label
      htmlFor="undefinedradio4"
      onClick={[Function bound handleClick]}>
      <div
        className="check" />
      項目5
    </label>
  </div>
  
</div>
`;
